### PR TITLE
Updated some dependencies to latest versions #203

### PR DESCRIPTION
**Pull request by stormwild**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -91,8 +91,8 @@
   ],
   "license": "MPL-2.0",
   "peerDependencies": {
-    "react": ">=16.x <=17.x",
-    "react-dom": ">=16.x <=17.x"
+    "react": "^18.0.0",
+    "react-dom": "^18.0.0"
```
[the remark this answers is not in the document]
Comment (stormwild) [reply package.json]: Updated with requested changes

**Review by nirajmchauhan — APPROVED** [review]

**Comment by nirajmchauhan** [comment]
Can this PR be merged? It will resolve this [258 issue](https://github.com/hashicorp/next-mdx-remote/issues/258) as well

**Comment by BRKalow** [comment]
Ah shucks, it looks like this is also running into https://github.com/hashicorp/next-mdx-remote/issues/254 now that react has been upgraded. 😞

**Comment by kwiat1990** [comment]
Hey, what’s the current state of this PR? Can we merge this one soon?

**Comment by stormwild** [comment]
@BRKalow @kwiat1990 Hi sorry hoping for guidance on any updates needed to resolve the failing checks. Haven't been able to check on the details on the failure yet. It seems there are issues related to peer-dependencies of dependencies? Hoping to find time to investigate this week.

**Comment by kyrisu** [comment]
There are 2 parts to the failed tests:

1. `Failed to compile` because of
```Module not found: Package path ./jsx-runtime.js```
This is a fairly simple fix - in React 18 it looks like that jsx-runtime is exported differently. Diff to fix:
```diff
diff --git a/rollup.config.js b/rollup.config.js
index 0829443..fe577ea 100644
--- a/rollup.config.js
+++ b/rollup.config.js
@@ -12,7 +12,7 @@ export default [
       dir: './dist',
       format: 'es',
     },
-    external: ['react', '@mdx-js/react', 'react/jsx-runtime.js'],
+    external: ['react', '@mdx-js/react', 'react/jsx-runtime'],
     plugins: [
       ts({
         tsconfig: './tsconfig.json',
diff --git a/src/index.tsx b/src/index.tsx
index 9b9f670..fc0ce27 100644
--- a/src/index.tsx
+++ b/src/index.tsx
@@ -1,6 +1,6 @@
 import './idle-callback-polyfill'
 import React, { useEffect, useState, useMemo } from 'react'
-import * as runtime from 'react/jsx-runtime.js'
+import * as runtime from 'react/jsx-runtime'
 import * as mdx from '@mdx-js/react'
 import { MDXRemoteSerializeResult } from './types'
```
We just need to remove the `.js` extension then run `npm run build` before running the tests.
2. Now that the build if progressing nicely the next one is the substring match:
```
 FAIL  __tests__/integration.test.tsx (6.092 s)
  hydration
    ✕ server rendered output (14 ms)
    ✓ rehydrates correctly in browser (720 ms)

  ● hydration › server rendered output

    expect(received).toContain(expected) // indexOf

    Expected substring: "<h1>foo</h1><h1>Headline</h1>
    <p>hello <!-- -->jeff</p><button>Count: <!-- -->0</button>
    <p class=\"context\">Context value: \"<!-- -->foo<!-- -->\"</p>
    <p>Some <strong class=\"custom-strong\">markdown</strong> content</p>
    <div class=\"alert alert-warning g-type-body\"><p>Alert</p></div>
    <div>I am a dynamic component.</div>"
    Received string:    "<h1>foo</h1><h1>Headline</h1>
    <!-- --><p>hello <!-- -->jeff<!-- --></p><button>Count: <!-- -->0<!-- --></button>
    <!-- --><p class=\"context\">Context value: \"<!-- -->foo<!-- -->\"<!-- --></p>
    <!-- --><p>Some <!-- --><strong class=\"custom-strong\">markdown</strong> content<!-- --></p>
    <!-- --><div class=\"alert alert-warning g-type-body\"><p>Alert</p></div>
    <!-- --><div>I am a dynamic component.</div>
    <!-- --><pre><code class=\"language-shell-session\">curl localhost
    </code></pre>
    <!-- --><p>This is more text.</p>
    <!-- --><div class=\"alert alert-warning g-type-body\"><p>&lt; client node IP &gt;:9999</p></div>
    <!-- --><p>\"Authorize &lt;GITHUB_USER&gt;\"</p>
    <!-- --><p>(support for version &lt;230)</p>
    <!-- --><h3>Some version &lt;= 1.3.x</h3>
    <!-- --><h4>metric.name.&lt;operation&gt;.&lt;mount&gt;</h4>
    <!-- --><p>&lt; 8ms</p>"

      23 |
      24 |     // server renders correctly
    > 25 |     expect($('#__next').html()).toContain(`<h1>foo</h1><h1>Headline</h1>
         |                                 ^
      26 | <p>hello <!-- -->jeff</p><button>Count: <!-- -->0</button>
      27 | <p class=\"context\">Context value: \"<!-- -->foo<!-- -->\"</p>
      28 | <p>Some <strong class=\"custom-strong\">markdown</strong> content</p>

      at Object.<anonymous> (__tests__/integration.test.tsx:25:33)

Test Suites: 1 failed, 1 passed, 2 total
Tests:       1 failed, 13 passed, 14 total
Snapshots:   11 passed, 11 total
Time:        6.236 s, estimated 33 s
Ran all test suites.
```
That actually looks correct to me - minus the empty comments (`<!-- -->`). I'm not sure why they are there, but seeing as they are in the tested substring, I'm assuming they are not the problem.

Modifying the test to look like this will work:
```javascript
  test('server rendered output', () => {
    const result = readOutputFile('basic', 'index')
    const $ = cheerio.load(result)

    // server renders correctly
    expect($('#__next').html().replace(/<!-- -->/g,'')).toContain(`<h1>foo</h1><h1>Headline</h1>
<p>hello jeff</p><button>Count: 0</button>
<p class=\"context\">Context value: \"foo\"</p>
<p>Some <strong class=\"custom-strong\">markdown</strong> content</p>
<div class=\"alert alert-warning g-type-body\"><p>Alert</p></div>
<div>I am a dynamic component.</div>`)
  })

```

Diff of the change

```diff
diff --git a/__tests__/integration.test.tsx b/__tests__/integration.test.tsx
index 67420f3..42a7461 100644
--- a/__tests__/integration.test.tsx
+++ b/__tests__/integration.test.tsx
@@ -22,9 +22,9 @@ describe('hydration', () => {
     const $ = cheerio.load(result)
 
     // server renders correctly
-    expect($('#__next').html()).toContain(`<h1>foo</h1><h1>Headline</h1>
-<p>hello <!-- -->jeff</p><button>Count: <!-- -->0</button>
-<p class=\"context\">Context value: \"<!-- -->foo<!-- -->\"</p>
+    expect($('#__next').html().replace(/<!-- -->/g,'')).toContain(`<h1>foo</h1><h1>Headline</h1>
+<p>hello jeff</p><button>Count: 0</button>
+<p class=\"context\">Context value: \"foo\"</p>
 <p>Some <strong class=\"custom-strong\">markdown</strong> content</p>
 <div class=\"alert alert-warning g-type-body\"><p>Alert</p></div>
 <div>I am a dynamic component.</div>`)
```

**Comment by stormwild** [comment]
@jescalan Closing resolved in https://github.com/hashicorp/next-mdx-remote/pull/267

**Pull request closed without merging**